src/main/flight/gps_rescue.c

```diff
@@ -264,24 +284,30 @@ static void rescueAttainPosition()
     // Speed and altitude controller internal variables
     static float previousSpeedError = 0;
     static int16_t speedIntegral = 0;
-    static float previousAltitudeError = 0;
-    static int16_t altitudeIntegral = 0;
+    int zVelocityError;
+    static int previousZVelocityError = 0;
+    static float zVelocityIntegral = 0;
+    static float scalingRate = 0;
+    static int16_t altitudeAdjustment = 0;
 
     if (rescueState.phase == RESCUE_INITIALIZE) {
         // Initialize internal variables each time GPS Rescue is started
         previousSpeedError = 0;
         speedIntegral = 0;
-        previousAltitudeError = 0;
-        altitudeIntegral = 0;
+        previousZVelocityError = 0;
+        zVelocityIntegral = 0;
+        altitudeAdjustment = 0;
     }
 
     // Point to home if that is in our intent
     if (rescueState.intent.crosstrack) {
         setBearing(rescueState.sensor.directionToHome);
     }
 
-    DEBUG_SET(DEBUG_RTH, 3, rescueState.failure); //Failure can change with no new GPS Data
-
+    if (debugMode == DEBUG_RTH) {
```

Comment:
No need to check this explicitly - `DEBUG_SET()` does this for you.

Reply:
Done